src/test/java/com/google/cloud/spanner/pgadapter/python/ITPsycopg2Test.java

```diff
@@ -90,4 +94,246 @@ public void testBasicInsert() throws IOException, InterruptedException {
     String expectedOutput = "1\n";
     assertEquals(expectedOutput, actualOutput);
   }
+
+  @Test
+  public void test2Select() throws IOException, InterruptedException {
+    String sql =
+        "Select "
+            + "col_bigint, "
+            + "col_bool , "
+            + "col_float8 , "
+            + "col_int , "
+            + "col_numeric , "
+            + "col_timestamptz , "
+            + "col_date , "
+            + "col_varchar "
+            + "from all_types where col_bigint = 2";
+    String actualOutput = executeWithoutParameters(testEnv.getPGAdapterPort(), sql, "query");
+    String expectedOutput =
+        "(2, True, 9.7, 10, Decimal('300.321'), datetime.datetime(2018, 3, 11, 10, 0, tzinfo=datetime.timezone.utc), datetime.date(2022, 10, 2), 'some string')\n";
+    assertEquals(expectedOutput, actualOutput);
+  }
+
+  @Test
+  public void test3Update() throws IOException, InterruptedException {
+    String sql =
+        "UPDATE all_types SET "
+            + "col_bool = 'off',"
+            + "col_float8 = 3.14, "
+            + "col_int = 20, "
+            + "col_numeric = 99.99, "
+            + "col_timestamptz = '2019-05-12 14:30:00'::timestamptz, "
+            + "col_date = '1989-09-02', "
+            + "col_varchar = 'new string' "
+            + "where col_bigint = 2";
+
+    String actualOutput = executeWithoutParameters(testEnv.getPGAdapterPort(), sql, "update");
+    String expectedOutput = "1\n";
+    assertEquals(expectedOutput, actualOutput);
+
+    sql =
+        "Select "
+            + "col_bigint, "
+            + "col_bool , "
+            + "col_float8 , "
+            + "col_int , "
+            + "col_numeric , "
+            + "col_timestamptz , "
+            + "col_date , "
+            + "col_varchar "
+            + "from all_types where col_bigint = 2";
+
+    actualOutput = executeWithoutParameters(testEnv.getPGAdapterPort(), sql, "query");
+    expectedOutput =
+        "(2, False, 3.14, 20, Decimal('99.99'), datetime.datetime(2019, 5, 12, 21, 30, tzinfo=datetime.timezone.utc), datetime.date(1989, 9, 2), 'new string')\n";
+    assertEquals(expectedOutput, actualOutput);
+  }
+
+  @Test
+  public void test4Delete() throws IOException, InterruptedException {
+    String sql = "DELETE FROM all_types where col_bigint = 2";
+
+    String actualOutput = executeWithoutParameters(testEnv.getPGAdapterPort(), sql, "update");
+    String expectedOutput = "1\n";
+    assertEquals(expectedOutput, actualOutput);
+
+    sql = "Select COUNT(*) from all_types where col_bigint = 2";
+
+    actualOutput = executeWithoutParameters(testEnv.getPGAdapterPort(), sql, "query");
+    expectedOutput = "(0,)\n";
```

Comment:
Each row in the result is returned as a tuple in Psycopg2. Since in this case there is only one column, the size of the tuple is 1.

In Python, tuples are the objects that are delimited by comma(,). So, even if the size of tuple is 1, we need to put a comma after the element. Parenthesis in tuple are optional.

For example, `t = (1, 2, 3)` will create a tuple and so will `t = 1, 2, 3`

For creating a tuple of size 1, `t = (1,)` and `t = 1,` will work but `t = (1)` will not work because tuples are identified by delimitation by comma and not by parenthesis, hence we need to put one comma after `1`

Reply:
Thanks for the elaborate explanation. That makes sense to me.